### PR TITLE
Loosen bounds in pick_appropriate_units

### LINKED ISSUE #24: 'pick_appropriate_units()': precision versus prettiness
I noticed recently that, due to the strict bounds checking in `pick_appropriate_units()` in `src/measurement.rs`, that the following code was not producing the output I was expecting:

```rust
extern crate measurements;
use measurements::*;

fn main() {
    let val : f64 = 1.0;
    println!("Mass of {0:.3}", Mass::from_kilograms(val));
    println!("Length of {0:.3}", Length::from_meters(val));
    println!("Volume of {0:.3}", Volume::from_litres(val));
}
```

prints the following:

```
Mass of 1000.000 g
Length of 100.000 cm
Volume of 1000.000 ml
```

whereas I had expected it to print:

```
Mass of 1.000 kg
Length of 1.000 m
Volume of 1.000 l
```

I figure the former is perhaps more precise, but seemed a bit ugly for display when dealing with units which didn't need that precision, particularly given that if I change `val` to be `1.00000000001`, I get the output I was expecting (due to precision formatting).

I've gone and updated my branch to use the latter by changing `value > 1.0 || value < -1.0` to `value >= 1.0 || value <= -1.0` in `pick_appropriate_units()`. If this is something that is generally more useful, or something to offer as an opt-in, I'd be happy to submit a PR.

### PR DESCRIPTION
**Pull request by alberdingk-thijm**
Change strict inequality to inequality for the inner loop of
pick_appropriate_units, so that if a value is set to 1.0, then
the unit returned follows a user's expectation.
Closes #24.

### CODE REVIEW

**Review by eldruin — COMMENTED** [review]
Thank you for this PR!
Could you add some unit tests for the behaviour?
Could you add an entry to the changelog?

**Comment by alberdingk-thijm** [comment]
@eldruin added the tests and changelog entry! Let me know if they conform to the project's style; happy to make edits.

File: tests/pick_appropriate.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    test_utils::assert_almost_eq,
+};
+
+#[test]
+fn appropriate_units_eq_1() {
+    let val : f64 = 1e0;
+    let mass = Mass::from_kilograms(val);
+    let (s, v) = mass.get_appropriate_units();
+    assert_eq!(s, "kg");
+    assert_almost_eq(val, v);
```
Comment (eldruin) [inline tests/pick_appropriate.rs:15]:
```suggestion
fn one_kg_keeps_unit() {
let value = 1.0;
    let mass = Mass::from_kilograms(value);
    let (unit, v) = mass.get_appropriate_units();
    assert_eq!(unit, "kg");
    assert_almost_eq(value, v);
```

File: tests/pick_appropriate.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+#[test]
+fn appropriate_units_neg_1() {
+    let val : f64 = -1e0;
+    let mass = Mass::from_kilograms(val);
+    let (s, v) = mass.get_appropriate_units();
+    assert_eq!(s, "kg");
+    assert_almost_eq(val, v);
```
Comment (eldruin) [inline tests/pick_appropriate.rs:24]:
```suggestion
fn minus_one_kg_keeps_unit() {
    let value = -1.0;
    let mass = Mass::from_kilograms(value);
    let (unit, v) = mass.get_appropriate_units();
    assert_eq!(unit, "kg");
    assert_almost_eq(value, v);
```

File: tests/pick_appropriate.rs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+#[test]
+fn appropriate_units_lt_1() {
+    // units will now be in grams
+    let val : f64 = 1e0 - 1e-3;
+    let mass = Mass::from_kilograms(val);
+    let (s, v) = mass.get_appropriate_units();
+    assert_eq!(s, "g");
+    assert_almost_eq(val * 1e3, v);
```
Comment (eldruin) [inline tests/pick_appropriate.rs:34]:
```suggestion
fn less_than_one_kg_changes_unit_to_grams() {
    let value = 1.0 - 0.001;
    let mass = Mass::from_kilograms(value);
    let (unit, v) = mass.get_appropriate_units();
    assert_eq!(unit, "g");
    assert_almost_eq(value * 1000, v);
```

File: tests/pick_appropriate.rs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+#[test]
+fn appropriate_units_gt_1() {
+    let val : f64 = 1e0 + 1e-3;
+    let mass = Mass::from_kilograms(val);
+    let (s, v) = mass.get_appropriate_units();
+    assert_eq!(s, "kg");
+    assert_almost_eq(val, v);
```
Comment (eldruin) [inline tests/pick_appropriate.rs:43]:
```suggestion
fn more_than_one_kg_keeps_unit() {
    let value = 1.001;
    let mass = Mass::from_kilograms(value);
    let (unit, v) = mass.get_appropriate_units();
    assert_eq!(unit, "kg");
    assert_almost_eq(value, v);
```

File: tests/pick_appropriate.rs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+#[test]
+fn appropriate_units_from_grams() {
+    let val : f64 = 1e3;
+    let mass = Mass::from_grams(val);
+    let (s, v) = mass.get_appropriate_units();
+    assert_eq!(s, "kg");
+    assert_almost_eq(val, v * 1e3);
```
Comment (eldruin) [inline tests/pick_appropriate.rs:52]:
```suggestion
fn one_thousand_grams_changes_unit_to_kg() {
    let value = 1e3;
    let mass = Mass::from_grams(value);
    let (unit, v) = mass.get_appropriate_units();
    assert_eq!(unit, "kg");
    assert_almost_eq(value, v * 1e3);
```

File: tests/pick_appropriate.rs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+#[test]
+fn appropriate_units_to_grams() {
+    let val : f64 = 1e-3;
+    let mass = Mass::from_kilograms(val);
+    let (s, v) = mass.get_appropriate_units();
+    assert_eq!(s, "g");
+    assert_almost_eq(val * 1e3, v);
```
Comment (eldruin) [inline tests/pick_appropriate.rs:61]:
```suggestion
fn one_gram_changes_unit_to_grams() {
    let value = 1e-3;
    let mass = Mass::from_kilograms(value);
    let (unit, v) = mass.get_appropriate_units();
    assert_eq!(unit, "g");
    assert_almost_eq(1.0, v);
```

**Review by eldruin — COMMENTED** [review]
Thank you for the changes!
I added some nitpick comments. The most important for me would be more descriptive test names.
In case you want to play around with macros, the code could also be simplified with something like this:
```rust
macro_rules! test_from_kg {
    ($name:ident, $value:expr, $unit:expr) => {
        #[test]
        fn $name {
            let mass = Mass::from_kilograms($value);
            let (unit, v) = mass.get_appropriate_units();
            assert_eq!(unit, $unit);
            assert_almost_eq($value, v);
        }
    };
}
```

**Comment by alberdingk-thijm** [comment]
Whoops, sorry for letting this languish. Thanks for the review, @eldruin, I've made the changes you suggested.

**Review by eldruin — APPROVED** [review]
Looks great to me. Thank you!

**Pull request merged**